zendesk/deco
# Add Zendesk Artifactory Settings and Core Libraries

:potable_water:

/cc @zendesk/sylvius

### Description

This adds the Core Zendesk Libraries to Deco. This includes configuration, core, database, global-uid, service, service, repository, and more. Tthe finatra version has changed since this must match core.

**Note**: When you run the server, you will get a warning that says: `ZENDESK_APP_NAME_IS_UNDEFINED","environment":"SCALA_ENV_IS_UNDEFINED`

To resolve this, we can add a deco configuration that defines these variables in the bin of docker-images. Something similar to [pravda's](https://github.com/zendesk/docker-images/blob/668161db86706a5b24f66a25cd3da78c722deb38/bin/pravda)

### How could this be manually tested?
* Run the server and make sure you can still hit the default endpoints.
* Try importing some of the core libraries in a Scala file, for example: `import com.zendesk.service._`


### Risks
* Low | The server may not start properly due to a broken build
